src/python/pants/help/help_info_extracter.py

```diff
@@ -522,6 +621,157 @@ def load() -> RuleInfo:
             }
         )
 
+    @classmethod
+    def get_api_type_infos(
+        cls, build_configuration: BuildConfiguration | None, union_membership: UnionMembership
+    ) -> LazyFrozenDict[str, PluginAPITypeInfo]:
+        if build_configuration is None:
+            return LazyFrozenDict({})
+
+        # The type narrowing achieved with the above `if` does not extend to the created closures of
+        # the nested functions in this body, so instead we capture it in a new variable.
+        bc = build_configuration
+
+        known_providers = cast(
+            "set[str]",
+            set(
+                chain.from_iterable(
+                    chain.from_iterable(cast(dict, providers).values())
+                    for providers in (
+                        bc.subsystem_to_providers,
+                        bc.target_type_to_providers,
+                        bc.rule_to_providers,
+                        bc.union_rule_to_providers,
+                    )
+                )
+            ),
+        )
+
+        def _find_provider(api_type: type) -> str:
```

Comment:
Is this logic duplicated from elsewhere? Is this to figure out the backend package?